consensus/polybft/signer/public.go

```diff
@@ -49,82 +29,86 @@ func (p *PublicKey) MarshalJSON() ([]byte, error) {
 }
 
 // UnmarshalJSON implements the json.Marshaler interface.
-func (p *PublicKey) UnmarshalJSON(b []byte) error {
-	var d []byte
+func (p *PublicKey) UnmarshalJSON(jsonBytes []byte) error {
+	var bytes []byte
 
-	err := json.Unmarshal(b, &d)
+	err := json.Unmarshal(jsonBytes, &bytes)
 	if err != nil {
 		return err
 	}
 
-	pk, err := UnmarshalPublicKey(d)
+	pub, err := UnmarshalPublicKey(bytes)
 	if err != nil {
 		return err
 	}
 
-	p.p = pk.p
+	p.g2 = pub.g2
 
 	return nil
 }
 
-// UnmarshalPublicKey reads the public key from the given byte array
-func UnmarshalPublicKey(raw []byte) (*PublicKey, error) {
-	if len(raw) == 0 {
-		return nil, errors.New("cannot unmarshal public key from empty slice")
-	}
-
-	p, err := bn254.NewG2().FromBytes(raw)
-	if err != nil {
-		return nil, err
-	}
-
-	return &PublicKey{p: p}, nil
-}
-
 // ToBigInt converts public key to 4 big ints
-func (p PublicKey) ToBigInt() [4]*big.Int {
+func (p *PublicKey) ToBigInt() [4]*big.Int {
 	blsKey := p.Marshal()
-	res := [4]*big.Int{
+	if len(blsKey) != 128 {
+		return [4]*big.Int{}
+	}
+
+	return [4]*big.Int{
 		new(big.Int).SetBytes(blsKey[32:64]),
 		new(big.Int).SetBytes(blsKey[0:32]),
 		new(big.Int).SetBytes(blsKey[96:128]),
 		new(big.Int).SetBytes(blsKey[64:96]),
 	}
+}
 
-	return res
+// UnmarshalPublicKey unmarshals bytes to public key
+func UnmarshalPublicKey(data []byte) (*PublicKey, error) {
+	g2 := new(bn256.G2)
+
+	if _, err := g2.Unmarshal(data); err != nil {
+		return nil, err
+	}
+
+	// check if it is the point at infinity
+	if g2.IsInfinity() {
+		return nil, errInfinityPoint
+	}
+
+	// check if not part of the subgroup
+	if !g2.InCorrectSubgroup() {
+		return nil, fmt.Errorf("incorrect subgroup")
+	}
+
+	return &PublicKey{g2: g2}, nil
 }
 
 // UnmarshalPublicKeyFromBigInt unmarshals public key from 4 big ints
 // Order of coordinates is [A.Y, A.X, B.Y, B.X]
 func UnmarshalPublicKeyFromBigInt(b [4]*big.Int) (*PublicKey, error) {
 	const size = 32
 
-	var pubKeyBuf []byte
```

Comment:
Was this change necessary?

Reply:
reverted https://github.com/0xPolygon/polygon-edge/pull/1161/commits/c39a660fed18b2128cee9bc0329b13d278d6c8b5